pandas/tseries/converter.py

```diff
@@ -216,7 +216,7 @@ def try_parse(values):
                 else:
                     values = [_dt_to_float_ordinal(x) for x in values]
             except Exception:
-                pass
+                values = _dt_to_float_ordinal(values)
```

Comment:
Allow me I do not understand converter dispatch logic. 

Might be unrelated to the PR, but is it possible that input can have a type not registered for ``DatetimeConverter`` like below condition? If so, it may better to box the above with ``try-except`` to return input as it is.

- https://github.com/jorisvandenbossche/pandas/blob/6b6b08e4fe13f1de310cd25a184625aed114a178/pandas/tseries/converter.py#L201